WMF Framework/Wikipedia.xcdatamodeld/Wikipedia 2.xcdatamodel/contents

```diff
@@ -0,0 +1,209 @@
+        <fetchIndex name="byErrorCodeAndCreatedDateIndex">
+            <fetchIndexElement property="errorCode" type="Binary" order="ascending"/>
+            <fetchIndexElement property="createdDate" type="Binary" order="ascending"/>
+        </fetchIndex>
+        <fetchIndex name="byErrorCodeAndDisplayTitleIndex">
+            <fetchIndexElement property="errorCode" type="Binary" order="ascending"/>
+            <fetchIndexElement property="displayTitle" type="Binary" order="ascending"/>
+        </fetchIndex>
+        <fetchIndex name="byUpdatedLocallyIndex">
+            <fetchIndexElement property="isUpdatedLocally" type="Binary" order="ascending"/>
+            <fetchIndexElement property="createdDate" type="Binary" order="ascending"/>
+        </fetchIndex>
+        <fetchIndex name="byListAndDeletedLocallyIndex">
+            <fetchIndexElement property="list" type="Binary" order="ascending"/>
+            <fetchIndexElement property="isDeletedLocally" type="Binary" order="ascending"/>
+        </fetchIndex>
+        <fetchIndex name="byReadingListEntryID">
+            <fetchIndexElement property="readingListEntryID" type="Binary" order="ascending"/>
+        </fetchIndex>
+        <fetchIndex name="byArticleKeyIndex">
+            <fetchIndexElement property="articleKey" type="Binary" order="ascending"/>
+        </fetchIndex>
+    </entity>
+    <entity name="TalkPage" representedClassName="TalkPage" syncable="YES" codeGenerationType="class">
+        <attribute name="key" attributeType="String" syncable="YES"/>
+        <attribute name="name" attributeType="String" syncable="YES"/>
+        <attribute name="revisionId" attributeType="Integer 64" minValueString="0" usesScalarValueType="YES" syncable="YES"/>
+        <relationship name="discussions" toMany="YES" deletionRule="Cascade" ordered="YES" destinationEntity="TalkPageDiscussion" inverseName="talkPage" inverseEntity="TalkPageDiscussion" syncable="YES"/>
+    </entity>
+    <entity name="TalkPageDiscussion" representedClassName="TalkPageDiscussion" syncable="YES" codeGenerationType="class">
+        <attribute name="title" attributeType="String" syncable="YES"/>
+        <relationship name="items" toMany="YES" deletionRule="Cascade" ordered="YES" destinationEntity="TalkPageDiscussionItem" inverseName="discussion" inverseEntity="TalkPageDiscussionItem" syncable="YES"/>
+        <relationship name="talkPage" maxCount="1" deletionRule="Deny" destinationEntity="TalkPage" inverseName="discussions" inverseEntity="TalkPage" syncable="YES"/>
+    </entity>
+    <entity name="TalkPageDiscussionItem" representedClassName="TalkPageDiscussionItem" syncable="YES" codeGenerationType="class">
+        <attribute name="depth" attributeType="Integer 16" defaultValueString="0" usesScalarValueType="YES" syncable="YES"/>
+        <attribute name="text" attributeType="String" syncable="YES"/>
```

Comment:
You could also keep the text in a file and store a reference as an attribute on `TalkPageDiscussionItem`, in some cases there could be a lot of text

Reply:
Good to know, I was unsure if it would be overkill so I didn't want to tackle it initially - I'll look into that after I flesh out the UI more.

Reply:
FYI planning to do this separately as a part of https://phabricator.wikimedia.org/T222721.